drivers/ipm/ipm_stm32_ipcc.c

```diff
@@ -0,0 +1,318 @@
+	struct stm32_ipcc_mbx_data *data = DEV_DATA(dev);
+	unsigned int value = 0;
+	u32_t mask, i;
+
+	mask = (~IPCC_ReadReg(data->hipcc, MR)) & IPCC_ALL_MR_RXO_CH_MASK;
+	mask &= IPCC_ReadOtherInstReg_SR(data->hipcc) & IPCC_ALL_SR_CH_MASK;
+
+	for (i = 0; i < data->num_ch; i++) {
+		if (!((1 << i) & mask)) {
+			continue;
+		}
+		LOG_DBG("%s channel = %x\r\n", __func__, i);
+		/* mask the channel Free interrupt  */
+		IPCC_DisableReceiveChannel(data->hipcc, i);
+
+		if (data->callback) {
+			/* Only one MAILBOX, id is unused and set to 0 */
+			data->callback(data->callback_ctx, i, &value);
+		}
+		/* clear status to acknoledge message reception */
+		IPCC_ClearFlag_CHx(data->hipcc, i);
+		IPCC_EnableReceiveChannel(data->hipcc, i);
+	}
+}
+
+static void stm32_ipcc_mailbox_tx_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct stm32_ipcc_mbx_data *data = DEV_DATA(dev);
+	u32_t mask, i;
+
+	mask = (~IPCC_ReadReg(data->hipcc, MR)) & IPCC_ALL_MR_TXF_CH_MASK;
+	mask = mask >> IPCC_C1MR_CH1FM_Pos;
+
+	mask &= IPCC_ReadReg_SR(data->hipcc) & IPCC_ALL_SR_CH_MASK;
+
+	for (i = 0; i <  data->num_ch; i++) {
+		if (!((1 << i) & mask)) {
+			continue;
+		}
+		LOG_DBG("%s channel = %x\r\n", __func__, i);
+		/* mask the channel Free interrupt */
+		IPCC_DisableTransmitChannel(data->hipcc, i);
+	}
+}
+
+static int stm32_ipcc_mailbox_ipm_send(struct device *d, int wait, u32_t id,
+				       const void *buff, int size)
+{
+	struct stm32_ipcc_mbx_data *data = d->driver_data;
+
+	ARG_UNUSED(wait);
+	ARG_UNUSED(buff);
+	ARG_UNUSED(size);
+
+	if (id >= data->num_ch) {
+		LOG_ERR("invalid id (%d)\r\n", id);
+		return  -EINVAL;
+	}
+
+	LOG_DBG("Send msg on channel %d\r\n", id);
+
+	/* Check that the channel is free (otherwise return error) */
+	if (IPCC_IsActiveFlag_CHx(data->hipcc, id)) {
+		LOG_DBG("Waiting for channel to be freed\r\n");
+		while (IPCC_IsActiveFlag_CHx(data->hipcc, id)) {
+			;
+		}
+	}
+	IPCC_EnableTransmitChannel(data->hipcc, id);
+	/* Inform A7 (either new message, or buf free) */
+	IPCC_SetFlag_CHx(data->hipcc, id);
+
+	return 0;
+}
+
+static int stm32_ipcc_mailbox_ipm_max_data_size_get(struct device *d)
+{
+	ARG_UNUSED(d);
+
+	/* Only a single 32-bit register available */
+	return 1;
+}
+
+static u32_t stm32_ipcc_mailbox_ipm_max_id_val_get(struct device *d)
+{
+	struct stm32_ipcc_mbx_data *data = DEV_DATA(d);
+
+	return data->num_ch - 1;
+}
+
+static void stm32_ipcc_mailbox_ipm_register_callback(struct device *d,
+						     ipm_callback_t cb,
+						     void *context)
+{
+	struct stm32_ipcc_mbx_data *data = DEV_DATA(d);
+
+	data->callback = cb;
+	data->callback_ctx = context;
+}
+
+static int stm32_ipcc_mailbox_ipm_set_enabled(struct device *dev, int enable)
+{
+	struct stm32_ipcc_mbx_data *data = DEV_DATA(dev);
+	u32_t i;
+
+	/* For now: nothing to be done */
+	LOG_DBG("%s %s mailbox\r\n", __func__, enable ? "enable" : "disable");
+	if (enable) {
+		/* Disable RX and TX interrupts */
+		IPCC_EnableIT_TXF(data->hipcc);
```

Comment:
TX interrupt could be enabled in `stm32_ipcc_mailbox_ipm_send`? I don't see a reason to enable it here.